src/app-routes/AppRoutes.tsx

```diff
@@ -8,19 +9,40 @@ import {
   QuestionPage,
   Result,
   Category,
+  Profile,
 } from "../pages";
+import { authContext } from "../context";
+import ProtectedRoutes from "./ProtectedRoutes";
 
 function AppRoutes() {
+  const {
+    authState: {
+      userDetails: { token },
+    },
+  } = useContext(authContext);
```

Comment:
Maybe you can use the useAuth hook here which you have already made.

Reply:
Thanks for suggestions, I have updated the code.